### DISCUSSION TITLE
Exec a custom pre-job script before executing a workflow

### DISCUSSION THREAD

**Discussion by Eldarrin** [actions/runner#2085] [Q&A] [answered]
Not sure if this is available but I have a simple script on my agent that does something like this:

```
#!/bin/bash
if [[ $(find ./ -type f -exec grep -Hn "github.com/notmyorg" {} \;) ]]; then
  echo "Found github.com/nomyorg in source code"
  exit 1
fi
```

So, if someone writes a workflow that tries to reach out of my org to somewhere else it'll bomb. But I want to make this script run before every job on every flow (there are a lot of repos). Also, can't really use webhooks because of InfoSec policy.

Any help appreciated.

Cheers, Andy

**Comment by lkfortuna** [comment, marked as answer]
Hi there - I am researching enabling this for hosted runners on Actions. We do have that capability for self hosted runners already the info is [here](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/running-scripts-before-or-after-a-job)

Were you wanting this capability on hosted or self hosted runners?

**Reply by Eldarrin** [reply]
It was actually on self-hosted runners, but as it needed to check the code required checkout, we implemented a workaround in the end with a custom checkout actions that checks the code before finishing the action as its on a secure server.

Cheers, Eld
